pkg/backend/httpstate/client/client.go

```diff
@@ -504,22 +504,6 @@ func (pc *Client) CompleteUpdate(ctx context.Context, update UpdateIdentifier, s
 		updateAccessToken(token), httpCallOptions{RetryAllMethods: true})
 }
 
-// AppendUpdateLogEntry appends the given entry to the indicated update's logs.
-func (pc *Client) AppendUpdateLogEntry(ctx context.Context, update UpdateIdentifier, kind string,
-	fields map[string]interface{}, token string) error {
-
-	req := apitype.AppendUpdateLogEntryRequest{
-		Kind:   kind,
-		Fields: fields,
-	}
-
-	// Retrying this POST operation could cause us to have multiple copies of the same log message for a given
-	// operation. The alternative, however, is worse. A failure in this call will fail the entire update, so we're
-	// forcing retry of these operations, at the expense of duplicated log messages in some cases.
-	return pc.updateRESTCall(ctx, "POST", getUpdatePath(update, "log"), nil, req, nil,
-		updateAccessToken(token), httpCallOptions{RetryAllMethods: true})
-}
-
 // RecordEngineEvent posts an engine event to the Pulumi service.
 func (pc *Client) RecordEngineEvent(
```

Comment:
Note that we should probably be retrying these calls. If the service returns a distinguished error code for duplicate events (e.g. a `409`), then we should be able to do this reliably.

Reply:
Will you follow up on this in another PR?

Reply:
Yes, sorry for missing it.

https://github.com/pulumi/pulumi/pull/2268/commits/ab4e437ee1b3ca2bf3bdd93de546913237af4e50